// @commercetools/platform-sdk/dist/platform-sdk.es.js
import { stringify as e } from 'querystring'
function r() {
  return (r =
    Object.assign ||
    function(e) {
      for (var r = 1; r < arguments.length; r++) {
        var s = arguments[r]
        for (var t in s)
          Object.prototype.hasOwnProperty.call(s, t) && (e[t] = s[t])
      }
      return e
    }).apply(this, arguments)
}
function s(e) {
  return null != e
}
function t(t) {
  var a,
    i,
    u,
    n = t.pathVariables,
    o = t.uriTemplate
  for (var h in n) o = o.replace('{' + h + '}', '' + n[h])
  return (
    o +
    ((a = t.queryParams || {}),
    (i = Object.keys(a).reduce(function(e, t) {
      var i,
        u = a[t]
      if (Array.isArray(u)) {
        var n,
          o = u.filter(s)
        return o.length ? r({}, e, (((n = {})[t] = o), n)) : e
      }
      return s(u) ? r({}, e, (((i = {})[t] = u), i)) : e
    }, {})),
    '' === (u = e(i)) ? '' : '?' + u)
  )
}
var a = (function() {
    function e(e, s) {
      ;(this.requestExecutor = s), (this.request = r({}, e, { uri: t(e) }))
    }
    var s = e.prototype
    return (
      (s.clientRequest = function() {
        return this.request
      }),
      (s.execute = function() {
        return this.requestExecutor(this.request)
      }),
      e
    )
  })(),
  i = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/api-clients/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/api-clients/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  u = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withId = function(e) {
        return new i({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/api-clients',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/api-clients',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  n = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/cart-discounts/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/cart-discounts/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/cart-discounts/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  o = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/cart-discounts/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/cart-discounts/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/cart-discounts/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  h = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withKey = function(e) {
        return new o({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withId = function(e) {
        return new n({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/cart-discounts',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/cart-discounts',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  g = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/carts/replicate',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  p = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/carts/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/carts/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/carts/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  l = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/carts/customer-id={customerId}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  d = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/carts/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/carts/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/carts/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  c = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.replicate = function() {
        return new g({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withCustomerId = function(e) {
        return new l({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withKey = function(e) {
        return new d({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withId = function(e) {
        return new p({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/carts',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/carts',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  y = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/categories/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/categories/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/categories/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  b = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/categories/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/categories/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/categories/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  q = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withKey = function(e) {
        return new b({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withId = function(e) {
        return new y({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/categories',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/categories',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  m = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/channels/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/channels/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/channels/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  U = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withId = function(e) {
        return new m({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/channels',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/channels',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  A = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/custom-objects/{container}/{key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/custom-objects/{container}/{key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  v = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/custom-objects/{container}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  f = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withContainerAndKey = function(e) {
        return new A({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withContainer = function(e) {
        return new v({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/custom-objects',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/custom-objects',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  T = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/customer-groups/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/customer-groups/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/customer-groups/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  x = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/customer-groups/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/customer-groups/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/customer-groups/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  R = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withKey = function(e) {
        return new x({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withId = function(e) {
        return new T({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/customer-groups',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/customer-groups',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  w = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/customers/email/confirm',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  j = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/customers/email-token',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  P = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/customers/password-token',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  K = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/customers/password',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  E = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/customers/password/reset',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  V = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/customers/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/customers/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/customers/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  k = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/customers/email-token={emailToken}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  D = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/customers/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/customers/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/customers/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  I = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate:
              '/{projectKey}/customers/password-token={passwordToken}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  C = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withPasswordToken = function(e) {
        return new I({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withEmailToken = function(e) {
        return new k({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.emailToken = function() {
        return new j({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.emailConfirm = function() {
        return new w({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.password = function() {
        return new K({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.passwordReset = function() {
        return new E({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.passwordToken = function() {
        return new P({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withKey = function(e) {
        return new D({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withId = function(e) {
        return new V({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/customers',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/customers',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  O = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/discount-codes/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/discount-codes/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/discount-codes/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  S = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withId = function(e) {
        return new O({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/discount-codes',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/discount-codes',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  G = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/extensions/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/extensions/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/extensions/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  L = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/extensions/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/extensions/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/extensions/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  z = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withKey = function(e) {
        return new L({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withId = function(e) {
        return new G({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/extensions',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/extensions',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  N = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/graphql',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/graphql' },
              null == e ? void 0 : e.headers
            ),
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  M = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate:
              '/{projectKey}/in-store/key={storeKey}/carts/replicate',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  W = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/in-store/key={storeKey}/carts/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/in-store/key={storeKey}/carts/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/in-store/key={storeKey}/carts/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  B = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate:
              '/{projectKey}/in-store/key={storeKey}/carts/customer-id={customerId}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  F = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate:
              '/{projectKey}/in-store/key={storeKey}/carts/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate:
              '/{projectKey}/in-store/key={storeKey}/carts/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate:
              '/{projectKey}/in-store/key={storeKey}/carts/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  H = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withCustomerId = function(e) {
        return new B({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withKey = function(e) {
        return new F({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.replicate = function() {
        return new M({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withId = function(e) {
        return new W({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/in-store/key={storeKey}/carts',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/in-store/key={storeKey}/carts',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  J = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate:
              '/{projectKey}/in-store/key={storeKey}/customers/email/confirm',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Q = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate:
              '/{projectKey}/in-store/key={storeKey}/customers/email-token',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  X = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate:
              '/{projectKey}/in-store/key={storeKey}/customers/password-token',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Y = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate:
              '/{projectKey}/in-store/key={storeKey}/customers/password',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Z = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate:
              '/{projectKey}/in-store/key={storeKey}/customers/password/reset',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  $ = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/in-store/key={storeKey}/customers/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/in-store/key={storeKey}/customers/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/in-store/key={storeKey}/customers/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  _ = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate:
              '/{projectKey}/in-store/key={storeKey}/customers/email-token={emailToken}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  ee = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate:
              '/{projectKey}/in-store/key={storeKey}/customers/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate:
              '/{projectKey}/in-store/key={storeKey}/customers/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate:
              '/{projectKey}/in-store/key={storeKey}/customers/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  re = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate:
              '/{projectKey}/in-store/key={storeKey}/customers/password-token={passwordToken}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  se = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withPasswordToken = function(e) {
        return new re({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withEmailToken = function(e) {
        return new _({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.emailToken = function() {
        return new Q({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.emailConfirm = function() {
        return new J({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.password = function() {
        return new Y({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.passwordReset = function() {
        return new Z({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.passwordToken = function() {
        return new X({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withKey = function(e) {
        return new ee({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withId = function(e) {
        return new $({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/in-store/key={storeKey}/customers',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/in-store/key={storeKey}/customers',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  te = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/in-store/key={storeKey}/login',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  ae = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/in-store/key={storeKey}/me/active-cart',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  ie = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/in-store/key={storeKey}/me/carts/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/in-store/key={storeKey}/me/carts/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/in-store/key={storeKey}/me/carts/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  ue = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withId = function(e) {
        return new ie({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/in-store/key={storeKey}/me/carts',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/in-store/key={storeKey}/me/carts',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  ne = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/in-store/key={storeKey}/me/orders/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  oe = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withId = function(e) {
        return new ne({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/in-store/key={storeKey}/me/orders',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/in-store/key={storeKey}/me/orders',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  he = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.carts = function() {
        return new ue({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.orders = function() {
        return new oe({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.activeCart = function() {
        return new ae({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      e
    )
  })(),
  ge = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/in-store/key={storeKey}/orders/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/in-store/key={storeKey}/orders/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/in-store/key={storeKey}/orders/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  pe = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate:
              '/{projectKey}/in-store/key={storeKey}/orders/order-number={orderNumber}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate:
              '/{projectKey}/in-store/key={storeKey}/orders/order-number={orderNumber}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate:
              '/{projectKey}/in-store/key={storeKey}/orders/order-number={orderNumber}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  le = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withOrderNumber = function(e) {
        return new pe({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withId = function(e) {
        return new ge({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/in-store/key={storeKey}/orders',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/in-store/key={storeKey}/orders',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  de = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate:
              '/{projectKey}/in-store/key={storeKey}/shipping-methods/matching-cart',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  ce = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.matchingCart = function() {
        return new de({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      e
    )
  })(),
  ye = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate:
              '/{projectKey}/in-store/key={storeKey}/shopping-lists/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate:
              '/{projectKey}/in-store/key={storeKey}/shopping-lists/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate:
              '/{projectKey}/in-store/key={storeKey}/shopping-lists/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  be = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate:
              '/{projectKey}/in-store/key={storeKey}/shopping-lists/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate:
              '/{projectKey}/in-store/key={storeKey}/shopping-lists/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate:
              '/{projectKey}/in-store/key={storeKey}/shopping-lists/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  qe = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withKey = function(e) {
        return new be({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withId = function(e) {
        return new ye({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/in-store/key={storeKey}/shopping-lists',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/in-store/key={storeKey}/shopping-lists',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  me = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.carts = function() {
        return new H({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.orders = function() {
        return new le({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.me = function() {
        return new he({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.customers = function() {
        return new se({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.login = function() {
        return new te({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.shippingMethods = function() {
        return new ce({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.shoppingLists = function() {
        return new qe({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      e
    )
  })(),
  Ue = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/inventory/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/inventory/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/inventory/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Ae = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withId = function(e) {
        return new Ue({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/inventory',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/inventory',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  ve = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/login',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  fe = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/me/active-cart',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Te = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/me/carts/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/me/carts/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/me/carts/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  xe = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/me/carts/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/me/carts/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/me/carts/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Re = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withKey = function(e) {
        return new xe({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withId = function(e) {
        return new Te({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/me/carts',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/me/carts',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  we = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/me/email/confirm',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  je = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/me/login',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Pe = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/me/orders/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Ke = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withId = function(e) {
        return new Pe({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/me/orders',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/me/orders',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Ee = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/me/password/reset',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Ve = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.reset = function() {
        return new Ee({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/me/password',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  ke = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/me/payments/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/me/payments/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/me/payments/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  De = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/me/payments/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/me/payments/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/me/payments/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Ie = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withKey = function(e) {
        return new De({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withId = function(e) {
        return new ke({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/me/payments',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/me/payments',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Ce = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/me/shopping-lists/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/me/shopping-lists/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/me/shopping-lists/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Oe = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/me/shopping-lists/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/me/shopping-lists/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/me/shopping-lists/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Se = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withId = function(e) {
        return new Ce({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withKey = function(e) {
        return new Oe({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/me/shopping-lists',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/me/shopping-lists',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Ge = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/me/signup',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Le = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.emailConfirm = function() {
        return new we({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.password = function() {
        return new Ve({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.signup = function() {
        return new Ge({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.login = function() {
        return new je({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.activeCart = function() {
        return new fe({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.carts = function() {
        return new Re({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.orders = function() {
        return new Ke({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.payments = function() {
        return new Ie({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.shoppingLists = function() {
        return new Se({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/me',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/me',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/me',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  ze = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/messages/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Ne = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withId = function(e) {
        return new ze({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/messages',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Me = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/orders/edits/{ID}/apply',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  We = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.apply = function() {
        return new Me({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/orders/edits/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/orders/edits/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/orders/edits/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Be = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/orders/edits/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/orders/edits/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/orders/edits/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Fe = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withKey = function(e) {
        return new Be({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withId = function(e) {
        return new We({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/orders/edits',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/orders/edits',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  He = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/orders/import',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Je = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/orders/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/orders/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/orders/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Qe = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/orders/order-number={orderNumber}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/orders/order-number={orderNumber}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/orders/order-number={orderNumber}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Xe = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.importOrder = function() {
        return new He({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withOrderNumber = function(e) {
        return new Qe({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.edits = function() {
        return new Fe({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withId = function(e) {
        return new Je({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/orders',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/orders',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Ye = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/payments/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/payments/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/payments/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Ze = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/payments/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/payments/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/payments/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  $e = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withKey = function(e) {
        return new Ze({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withId = function(e) {
        return new Ye({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/payments',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/payments',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  _e = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/product-discounts/matching',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  er = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/product-discounts/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/product-discounts/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/product-discounts/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  rr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/product-discounts/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/product-discounts/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/product-discounts/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  sr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.matching = function() {
        return new _e({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withKey = function(e) {
        return new rr({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withId = function(e) {
        return new er({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/product-discounts',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/product-discounts',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  tr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/product-projections/search',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/x-www-form-urlencoded' },
              null == e ? void 0 : e.headers
            ),
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/product-projections/search',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  ar = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/product-projections/suggest',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  ir = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/product-projections/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  ur = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/product-projections/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  nr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.search = function() {
        return new tr({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.suggest = function() {
        return new ar({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withKey = function(e) {
        return new ur({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withId = function(e) {
        return new ir({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/product-projections',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  or = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/product-types/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/product-types/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/product-types/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  hr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/product-types/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/product-types/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/product-types/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  gr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withKey = function(e) {
        return new hr({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withId = function(e) {
        return new or({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/product-types',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/product-types',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  pr = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/products/{ID}/images',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  lr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.images = function() {
        return new pr({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/products/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/products/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/products/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  dr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/products/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/products/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/products/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  cr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withKey = function(e) {
        return new dr({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withId = function(e) {
        return new lr({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/products',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/products',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  yr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/reviews/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/reviews/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/reviews/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  br = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/reviews/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/reviews/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/reviews/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  qr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withKey = function(e) {
        return new br({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withId = function(e) {
        return new yr({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/reviews',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/reviews',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  mr = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/shipping-methods/matching-cart',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Ur = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/shipping-methods/matching-location',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Ar = (function() {
    function e(e) {
      this.args = e
    }
    return (
      (e.prototype.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/shipping-methods/matching-orderedit',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  vr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/shipping-methods/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/shipping-methods/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/shipping-methods/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  fr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/shipping-methods/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/shipping-methods/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/shipping-methods/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Tr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withKey = function(e) {
        return new fr({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.matchingCart = function() {
        return new mr({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.matchingOrderedit = function() {
        return new Ar({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.matchingLocation = function() {
        return new Ur({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withId = function(e) {
        return new vr({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/shipping-methods',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/shipping-methods',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  xr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/shopping-lists/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/shopping-lists/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/shopping-lists/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Rr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/shopping-lists/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/shopping-lists/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/shopping-lists/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  wr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withKey = function(e) {
        return new Rr({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withId = function(e) {
        return new xr({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/shopping-lists',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/shopping-lists',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  jr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/states/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/states/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/states/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Pr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/states/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/states/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/states/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Kr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withKey = function(e) {
        return new Pr({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withId = function(e) {
        return new jr({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/states',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/states',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Er = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/stores/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/stores/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/stores/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Vr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/stores/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/stores/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/stores/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  kr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withKey = function(e) {
        return new Vr({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withId = function(e) {
        return new Er({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/stores',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/stores',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Dr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/subscriptions/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/subscriptions/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/subscriptions/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Ir = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/subscriptions/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/subscriptions/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/subscriptions/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Cr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withKey = function(e) {
        return new Ir({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withId = function(e) {
        return new Dr({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/subscriptions',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/subscriptions',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Or = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/tax-categories/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/tax-categories/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/tax-categories/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Sr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/tax-categories/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/tax-categories/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/tax-categories/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Gr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withKey = function(e) {
        return new Sr({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withId = function(e) {
        return new Or({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/tax-categories',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/tax-categories',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Lr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/types/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/types/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/types/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  zr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/types/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/types/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/types/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Nr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withKey = function(e) {
        return new zr({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withId = function(e) {
        return new Lr({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/types',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/types',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Mr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/zones/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/zones/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/zones/{ID}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Wr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/zones/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/zones/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      (s.delete = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'DELETE',
            uriTemplate: '/{projectKey}/zones/key={key}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Br = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.withKey = function(e) {
        return new Wr({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.withId = function(e) {
        return new Mr({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}/zones',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
            queryParams: null == e ? void 0 : e.queryArgs,
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}/zones',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            queryParams: null == e ? void 0 : e.queryArgs,
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Fr = (function() {
    function e(e) {
      this.args = e
    }
    var s = e.prototype
    return (
      (s.categories = function() {
        return new q({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.carts = function() {
        return new c({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.cartDiscounts = function() {
        return new h({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.channels = function() {
        return new U({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.customers = function() {
        return new C({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.customerGroups = function() {
        return new R({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.customObjects = function() {
        return new f({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.discountCodes = function() {
        return new S({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.graphql = function() {
        return new N({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.inventory = function() {
        return new Ae({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.login = function() {
        return new ve({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.messages = function() {
        return new Ne({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.orders = function() {
        return new Xe({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.payments = function() {
        return new $e({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.products = function() {
        return new cr({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.productDiscounts = function() {
        return new sr({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.productProjections = function() {
        return new nr({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.productTypes = function() {
        return new gr({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.reviews = function() {
        return new qr({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.shippingMethods = function() {
        return new Tr({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.shoppingLists = function() {
        return new wr({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.states = function() {
        return new Kr({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.subscriptions = function() {
        return new Cr({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.taxCategories = function() {
        return new Gr({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.types = function() {
        return new Nr({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.zones = function() {
        return new Br({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.me = function() {
        return new Le({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.extensions = function() {
        return new z({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.apiClients = function() {
        return new u({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.stores = function() {
        return new kr({
          pathArgs: r({}, this.args.pathArgs),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.inStoreKeyWithStoreKeyValue = function(e) {
        return new me({
          pathArgs: r({}, this.args.pathArgs, e),
          executeRequest: this.args.executeRequest,
          baseUri: this.args.baseUri,
        })
      }),
      (s.get = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'GET',
            uriTemplate: '/{projectKey}',
            pathVariables: this.args.pathArgs,
            headers: r({}, null == e ? void 0 : e.headers),
          },
          this.args.executeRequest
        )
      }),
      (s.post = function(e) {
        return new a(
          {
            baseUri: this.args.baseUri,
            method: 'POST',
            uriTemplate: '/{projectKey}',
            pathVariables: this.args.pathArgs,
            headers: r(
              { 'Content-Type': 'application/json' },
              null == e ? void 0 : e.headers
            ),
            body: null == e ? void 0 : e.body,
          },
          this.args.executeRequest
        )
      }),
      e
    )
  })(),
  Hr = (function() {
    function e(e) {
      var r
      ;(this.executeRequest = e.executeRequest),
        (this.baseUri =
          null != (r = e.baseUri)
            ? r
            : 'https://api.europe-west1.gcp.commercetools.com')
    }
    return (
      (e.prototype.withProjectKey = function(e) {
        return new Fr({
          pathArgs: r({}, e),
          executeRequest: this.executeRequest,
          baseUri: this.baseUri,
        })
      }),
      e
    )
  })(),
  Jr = function(e, r) {
    return r && 0 != r.length ? Xr(r.reduce(Qr), e) : e
  }
function Qr(e, r) {
  return function(s, t) {
    return e(s, Xr(r, t))
  }
}
function Xr(e, r) {
  return function(s) {
    return e(s, r)
  }
}
function Yr(e, r) {
  return new Hr({ executeRequest: e.execute, baseUri: r })
}
export {
  Hr as ApiRoot,
  Yr as createApiBuilderFromCtpClient,
  Jr as createExecutorFromMiddlewares,
}
